x/interchainqueries/types/genesis.go

```diff
@@ -10,5 +15,40 @@ func DefaultGenesis() *GenesisState {
 // Validate performs basic genesis state validation returning an error upon any
 // failure.
 func (gs GenesisState) Validate() error {
-	return gs.Params.Validate()
+	err := gs.Params.Validate()
+	if err != nil {
+		return err
+	}
+	seenIDs := map[uint64]bool{}
+
+	for _, val := range gs.GetRegisteredQueries() {
+		if seenIDs[val.Id] {
+			return sdkerrors.Wrapf(ErrInvalidQueryID, "duplicate query id: %d", val.Id)
+		}
+		seenIDs[val.Id] = true
+
+		_, err = sdk.AccAddressFromBech32(val.Owner)
+		if err != nil {
+			return sdkerrors.Wrapf(err, "Invalid owner address (%s)", err)
+		}
+
+		if val.QueryType == "" {
```

Comment:
please consider refactoring like

```go
switch val.QueryType {
    case "tx":
        // validate tx
    case "kv":
        // validate kv
    default:
        // return error like "unexpected query type %s"
}
```

this is more pretty imo and otherwise query type "blah-blah" is a valid query type

and please reuse "tx" and "kv" consts from `x/interchainqueries/types/types.go`